### PR TITLE
✨Add IPAM provider support

### LINKED ISSUE #367: Support of cluster api IPAM provider
**User Story**

As an operator I would like cluster api IPAM provider to be supported for beeing able to deploy all providers I need using cluster api operator only (and not use clusterctl for IPAM provider and cluster api operator for other providers).

**Detailed Description**

Cluster api IPAM provider is a specific provider supported by cluster api but currently not supported in cluster api operator for beeing deployed. It will be great if you can add support for this provider too.

**Anything else you would like to add:**

[Miscellaneous information that will assist in solving the issue.]

/kind feature

### PR DESCRIPTION
**Pull request by Fedosin**
**What this PR does / why we need it**:

This PR adds a new provider type - `IPAMProvider`, that will be used to manage Cluster API [IPAM providers](https://cluster-api.sigs.k8s.io/reference/providers.html#ip-address-management-ipam).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #367

### CODE REVIEW

**Review by Danil-Grigorev — COMMENTED** [review]
/lgtm

File: api/v1alpha2/ipamprovider_types.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (furkatgofurov7) [inline api/v1alpha2/ipamprovider_types.go:2]: nit: header year has to be updated here in other places

**Review by furkatgofurov7 — APPROVED** [review]

File: api/v1alpha2/ipamprovider_types.go (hunk 1 above)
Comment (Fedosin) [reply api/v1alpha2/ipamprovider_types.go:2]: Idk... the code was written last year and hasn't been updated after, so I think it is correct to put 2023 there.

**Pull request merged**